### PR TITLE
Elements api init

### PR DESCRIPTION
**Pull request by pychik**
Related to #27.

### CODE REVIEW

**Comment by pychik** [comment]
- elements methods performed.
- small fixes in testcontainers because of browsers issue,
- syntax issue in test_navigation fixed

- also  in frames def switch_to():
     - need to divide this method to 2 methods switch_to_index, switch_to_element because Gvido says: **this is the way**
     -  **FrameId** bug of swagger: webdriver 's not working with  FrameId model- so had to make some hardcode

File: lightning/webelement.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+        self._session_id = session_id
+
+    def accessibility(self) -> Accessibility:
+        _accessibility = Accessibility(elements_instance=self._elements_instance,
+                                       session_id=self.session_id,
+                                       element_id=self.element_id)
+        return _accessibility
+
+    @property
+    def element_id(self) -> str:
```
Comment (vania-pooh) [inline lightning/webelement.py]: Rename to just `id()`.

File: lightning/webelement.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    def __repr__(self):
+        _c = dict(role=self.role, label=self.label)
+        return repr(_c)
+
+
+class WebElement(dict):
+
+    def __init__(self, elements_instance: ElementsApi, element_id: str, session_id: str):
+        self._elements_instance = elements_instance
+        self._element_id = element_id
```
Comment (vania-pooh) [inline lightning/webelement.py]: Renamed `_element_id` -> `_id`.

File: lightning/webelement.py [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+        _web_element = self._elements_instance.element_clear(session_id=self.session_id,
+                                                             element_id=self.element_id,
+                                                             body=EmptyRequest())
+        return self
+
+    def find_all(self, locator: Locator) -> list[WebElement]:
+        _body = FindElementRequest(using=locator.strategy, value=locator.expression)
+        _response: list[FindElementResponseValue] = self._elements_instance.find_elements(session_id=self.session_id,
+                                                                                          body=_body).value
+        # ? raise smth if len(_response) == 0
```
Comment (vania-pooh) [inline lightning/webelement.py]: Not needed. If response is empty list we return an empty list of elements.

File: lightning/webelement.py [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+            _web_element = WebElement(elements_instance=self._elements_instance,
+                                      element_id=_response.element_6066_11e4_a52e_4f735466cecf,
+                                      session_id=self.session_id)
+            return _web_element
+        except Exception as e:
+            raise WebDriverException() from e
+
+    @property
+    def selected(self) -> bool:
+        _response = self._elements_instance.is_element_selected(session_id=self.session_id, element_id=self.element_id)
```
Comment (vania-pooh) [inline lightning/webelement.py]: Probably can be shortened to one line:
```
return self._elements_instance.is_element_selected(session_id=self.session_id, element_id=self.element_id).value
```

File: lightning/webelement.py [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+                                                                        element_id=self.element_id).value
+        return Position(x=int(_rect_response.x), y=int(_rect_response.y))
+
+    @property
+    def size(self) -> Size:
+        _rect_response: Rect = self._elements_instance.get_element_rect(session_id=self.session_id,
+                                                                        element_id=self.element_id).value
+        return Size(width=int(_rect_response.width), height=int(_rect_response.height))
+
+    def element_property(self, name: str) -> Optional[str]:
```
Comment (vania-pooh) [inline lightning/webelement.py:169]: Should be just `property`.

File: lightning/webelement.py [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+        return self._session_id
+
+    def click(self) -> WebElement:
+        _web_element = self._elements_instance.element_click(session_id=self.session_id,
+                                                             element_id=self.element_id,
+                                                             body=EmptyRequest())
+        return self
+
+    def clear(self) -> WebElement:
+        _web_element = self._elements_instance.element_clear(session_id=self.session_id,
```
Comment (vania-pooh) [inline lightning/webelement.py:106]: Every such call can throw ApiException. Need to catch it and throw WebDriverException instead with cause. To not clutter the code with boilterplate try ... catch you can implement a function like this: https://github.com/aerokube/lightning-java/blob/master/lightning/src/main/java/com/aerokube/lightning/StdWebDriver.java#L122-L130

File: lightning/webelement.py [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+                                                             element_id=self.element_id)
+        return self
+
+    def screenshot(self) -> bytes:
+        _api_client = self._elements_instance.api_client
+        _screenshot_instance = ScreenshotsApi(_api_client)
+        return b64decode(_screenshot_instance.take_screenshot(session_id=self.session_id).value)
+
+    def __repr__(self):
+        """This method we use for representation of our WebElement if user decided to print it"""
```
Comment (vania-pooh) [inline lightning/webelement.py]: Useless docs. Remove it.

**Review by vania-pooh — CHANGES_REQUESTED** [review]
Need to do some minor changes.

File: lightning/webelement.py (hunk 6 above)
Comment (pychik) [reply lightning/webelement.py:106]: not sure 
fixed in 990528b5d10e02f0199fddd9a570dc33e95fa248

File: lightning/webelement.py (hunk 5 above)
Comment (pychik) [reply lightning/webelement.py:169]: can't use inner decorator **property** name

File: lightning/webelement.py (hunk 4 above)
Comment (pychik) [reply lightning/webelement.py]: fixed in 990528b5d10e02f0199fddd9a570dc33e95fa248

File: lightning/webelement.py (hunk 3 above)
Comment (pychik) [reply lightning/webelement.py]: updated in 990528b5d10e02f0199fddd9a570dc33e95fa248

File: lightning/webelement.py (hunk 2 above)
Comment (pychik) [reply lightning/webelement.py]: fixed in 990528b5d10e02f0199fddd9a570dc33e95fa248

File: lightning/webelement.py (hunk 1 above)
Comment (pychik) [reply lightning/webelement.py]: Pycharm didn't like it,  but it works fine

fixed in 990528b5d10e02f0199fddd9a570dc33e95fa248

File: lightning/webelement.py (hunk 5 above)
Comment (vania-pooh) [reply lightning/webelement.py:169]: Ok.

File: lightning/webelement.py [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+        return self
+
+    def clear(self) -> WebElement:
+        try:
+            _web_element = self._elements_instance.element_clear(session_id=self.session_id,
+                                                                 element_id=self.id,
+                                                                 body=EmptyRequest())
+
+            return self
+        except ApiException as e:
```
Comment (vania-pooh) [inline lightning/webelement.py]: You need to apply the same to all methods, not only this one. To not copy-paste a lot - use decorator for that: https://stackoverflow.com/questions/15572288/general-decorator-to-wrap-try-except-in-python

File: lightning/elements.py [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+                                  id=_response.element_6066_11e4_a52e_4f735466cecf,
+                                  session_id=self.session_id)
+        return _web_element
+
+    def find_all(self, locator: Locator) -> list[WebElement]:
+        _body = FindElementRequest(using=locator.strategy, value=locator.expression)
+        _response: list[FindElementResponseValue] = self._elements_instance.find_elements(session_id=self.session_id,
+                                                                                          body=_body).value
+
+        # ? raise smth if len(_response) == 0
```
Comment (vania-pooh) [inline lightning/elements.py]: Still present.

**Review by vania-pooh — CHANGES_REQUESTED** [review]

**Comment by pychik** [comment]
def **switch_to** refactored as single method with 2 ways of handling arguments

**Review by vania-pooh — APPROVED** [review]

**Pull request merged**